packages/edit-site/src/components/editor/index.js

```diff
@@ -173,6 +170,17 @@ export default function EditSiteEditor( { isLoading } ) {
 		[ history, createSuccessNotice ]
 	);
 
+	// Replace the title and icon displayed in the DocumentBar when there's an overlay visible.
+	const { setEditorContentOverlayTitleAndIcon } = unlock(
+		useDispatch( editorStore )
+	);
+	const { title, icon } = getEditorCanvasContainerTitleAndIcon(
+		editorCanvasContainerView
+	);
+	useEffect( () => {
+		setEditorContentOverlayTitleAndIcon( title, icon );
+	}, [ title, icon, setEditorContentOverlayTitleAndIcon ] );
```

Comment:
This basically replaces the "title" prop with an action API. Do you think it's a better API? Why?

If we're to keep this, I guess we should remove the "title" prop from the Editor component (and its children)

Reply:
@youknowriad Thanks for taking look

> This basically replaces the "title" prop with an action API. Do you think it's a better API? Why?

I did it this way because I was unsure of the backcompat restraints on the `@wordpress/editor` components, and this way keeps all API changes private.

Alternatively we could make the following modifications

- Add an `icon` prop to `Editor` (from `@wordpress/editor`) so that `EditSiteEditor` can use it to override the icon (in addition to the title override it's using now) 
- Add title and icon props to the `DocumentBar` component to override the title and icon inferred from the currently open document or template, so that we can display the title and icon overrides from `EditSiteEditor`  inside the DocumentBar when an overlay is visible (rather than just replacing the component with a string, like we do now)

Given the [plans to refactor EditorCanvasContainer](https://github.com/WordPress/gutenberg/issues/62216) for Style Book and Style Revisions I'm hesitant to add new props to `@wordpress/editor` components just for the sake of the site editor. On the other hand, those implementing custom editors might appreciate the title/icon override props, so it could make sense to keep them around. What do you think?

Reply:
To be honest, neither solution satisfy me completely :P 

 - I like the simplicity of the props but don't like the fact that they need to be passed down to the components.
 - I don't like the complexity of the actions/selectors but they don't need to be passed down to components.

Worth nothing that the components we're talking about are also private AFAIK. So maybe for now, I have a small preference towards just adding an icon prop.

Reply:
> To be honest, neither solution satisfy me completely ... So maybe for now, I have a small preference towards just adding an icon prop.

I feel similarly! If nothing else, using props is a little less code overall. I'll rework the PR to use props and we can see how that feels.

Reply:
By the way, `DocumentBar` is exported from `@wordpress/editor`, so adding props to the component does change the public API of the package.